library/index.c

```diff
@@ -530,10 +530,14 @@ int eblob_generate_sorted_index(struct eblob_backend *b, struct eblob_base_ctl *
 	struct eblob_map_fd src, dst;
 	int fd, err, len;
 	char *file, *dst_file;
+	int cache_is_empty = 0;
 
 	if (b == NULL || bctl == NULL)
 		return -EINVAL;
 
+	/* check that eblob cache is empty. If cache is empty we can skip iterating and flushing it */
+	cache_is_empty = eblob_cache_empty(b);
+
```

Comment:
"Empty cache at the start of index sorting means there is no key from the blob in the cache and after sorting it is useless to flush blob's keys from the cache."

It is not about flush, but binlog - one has to apply it even if there were now cached keys - another user may overwrite or remove keys which are being sorted.

If cache is empty there is no need to interate over index file and flush its entries.